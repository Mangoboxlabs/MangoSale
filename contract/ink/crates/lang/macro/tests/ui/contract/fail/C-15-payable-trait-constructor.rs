use ink_lang as ink;

#[ink::trait_definition]
pub trait Constructor {
    #[ink(constructor)]
    fn constructor() -> Self;
}

#[ink::contract]
mod noop {
    #[ink(storage)]
    pub struct Noop {}

    impl Constructor for Noop {
        #[ink(constructor, payable)]
        fn constructor() -> Self {
            Self {}
        }
    }

    impl Noop {
        #[ink(message)]
        pub fn noop(&self) {}
    }
}

fn main() {}
