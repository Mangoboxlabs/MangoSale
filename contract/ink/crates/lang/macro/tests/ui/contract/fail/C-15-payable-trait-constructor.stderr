error: encountered conflicting ink! attribute argument
  --> $DIR/C-15-payable-trait-constructor.rs:15:28
   |
15 |         #[ink(constructor, payable)]
   |                            ^^^^^^^

error: constructors are implicitly payable
  --> $DIR/C-15-payable-trait-constructor.rs:15:28
   |
15 |         #[ink(constructor, payable)]
   |                            ^^^^^^^
